numpy/_core/src/multiarray/multiarraymodule.c

```diff
@@ -99,16 +99,42 @@ _umath_strings_richcompare(
         PyArrayObject *self, PyArrayObject *other, int cmp_op, int rstrip);
 
 
-static PyObject *
-set_legacy_print_mode(PyObject *NPY_UNUSED(self), PyObject *args)
-{
-    if (!PyArg_ParseTuple(args, "i", &npy_thread_unsafe_state.legacy_print_mode)) {
-        return NULL;
+NPY_NO_EXPORT int
+get_legacy_print_mode(void) {
+    /*
+     * Get the C value of the legacy printing mode.
+     * accessible from C. For simplicity the mode is encoded as an
+     * integer where INT_MAX means no legacy mode, and '113'/'121'/'125'
+     * means 1.13/1.21/1.25 legacy mode; and 0 maps to INT_MAX. We can
+     * upgrade this if we have more complex requirements in the future.
+     */
+    PyObject *val = NULL;
+    PyContextVar_Get(npy_static_pydata._format_options, NULL, &val);
+    if (val == NULL) {
+        PyErr_SetString(PyExc_SystemError,
+                        "NumPy internal error: unable to get _format_options "
+                        "context variable");
+        return -1;
     }
-    if (!npy_thread_unsafe_state.legacy_print_mode) {
-        npy_thread_unsafe_state.legacy_print_mode = INT_MAX;
+    PyObject *legacy_print_mode = NULL;
+    if (PyDict_GetItemStringRef(val, "legacy", &legacy_print_mode) == -1) {
+        return -1;
     }
-    Py_RETURN_NONE;
+    Py_DECREF(val);
+    if (legacy_print_mode == NULL) {
+        PyErr_SetString(PyExc_SystemError,
+                        "NumPy internal error: unable to get legacy print "
+                        "mode");
+    }
+    long long ret = PyLong_AsLongLong(legacy_print_mode);
+    Py_DECREF(legacy_print_mode);
+    // TODO: check for overflow, which should never happen
+    if (ret == LLONG_MAX) {
+        return INT_MAX;
+    }
```

Comment:
I guess you get away with it :), since there are only two choices and 32bit matches with int...
But slightly more correct: `sys.maxsize` matches `PyLong_AsSsize_t` and not `LLONG_MAX`.  Not sure there is an `SSIZE_T_MAX` macro (presumably), although `ret > INT_MAX: return INT_MAX` seems just as well.

And that removes that "TODO" comment, I guess.  (or I guess return an `Py_ssize_t` or just change the max value we use :))